Session_1/module_1.py

```diff
@@ -2,35 +2,77 @@
 
 
 def task_1(array: List[int], target: int) -> List[int]:
-    """
-    Write your code below
-    """
-    pass
+    # Initialize an empty dictionary  that saves complementing numbers
+    storage = {}
+
+    # go through all the numbers in the array and fild its complementing number
+    for i in array:  # 2 3 6 7 8 9      10
+        comp_num = target - i
+        # check if the comp_num is already in the dictionary
+        if comp_num in storage:
+            # if yes, return the pair
+            return [comp_num, i]
+        # if no, add to disct
+        storage[i] = True
+    # If no pair is found, returning emply list
+    return []
 
 
 def task_2(number: int) -> int:
-    """
-    Write your code below
-    """
-    pass
+    if number < 0:
+        number = number * -1
+        sign = -1
+    else:
+        sign = 1
+
+    mirror = 0
+    while number > 0:
+        a = number % 10
+        number = number // 10
+        mirror = mirror * 10 + a
+
+    return mirror * sign
 
 
 def task_3(array: List[int]) -> int:
-    """
-    Write your code below
-    """
-    pass
+    for i in range(len(array)):
+        if array[i] in array[:i]:
```

Comment:
Solution is correct, but it will run a long time for long arrays.
Operation `array[i] in array[:i]` goes through the list `array[:i]` and checks if each element is equal to `array[i]`. So the solution will run in O(n^2) where `n` is the length of the list. In practice we try to avoid it.